src/journey-maps/angular/services/source/route-source-service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@angular/core';
+import { Feature, FeatureCollection } from 'geojson';
+import { GeoJSONSource, Map as MaplibreMap } from 'maplibre-gl';
+
+import { ROKAS_ROUTE_SOURCE } from '../constants';
+import { SbbMapSelectionEvent } from '../map/events/map-selection-event';
+import { SBB_EMPTY_FEATURE_COLLECTION } from '../map/map-service';
+
+import { needsFloorChange, updateWalkFloor } from './util/walk-floor-updater';
+
+@Injectable({ providedIn: 'root' })
+export class SbbRouteSourceService {
+  private _data: FeatureCollection;
+
+  updateRoute(
+    map: MaplibreMap,
+    mapSelectionEventService: SbbMapSelectionEvent,
+    routeFeatureCollection: FeatureCollection = SBB_EMPTY_FEATURE_COLLECTION
+  ): void {
+    const source = map.getSource(ROKAS_ROUTE_SOURCE) as GeoJSONSource;
+    source.setData(routeFeatureCollection);
+    map.removeFeatureState({ source: ROKAS_ROUTE_SOURCE });
+    if (routeFeatureCollection.features?.length) {
+      map.once('idle', () =>
+        mapSelectionEventService.initSelectedState(map, routeFeatureCollection.features, 'ROUTE')
+      );
+    }
+  }
+
+  updateTransferV2(
```

Comment:
I would probably put the transfer (footpath) related stuff into the TransferService.

Reply:
Done ✅